sql/analyzer/hoist_select_exists.go

```diff
@@ -231,11 +255,13 @@ func decorrelateOuterCols(e *plan.Subquery, scopeLen int, aliasDisambig *aliasDi
 	if err != nil {
 		return nil, err
 	}
+
+	// add good aliases to outsideAliases
 	conflicts, nonConflicted := outsideAliases.findConflicts(nodeAliases)
 	for _, goodAlias := range nonConflicted {
 		target, ok := nodeAliases[goodAlias]
 		if !ok {
-			return nil, fmt.Errorf("node alias %s is not in nodeAliases", goodAlias)
+			return nil, fmt.Errorf("good node alias %s is not in nodeAliases", goodAlias)
```

Comment:
> We need terminology to be precise and unambiguous. What is a "good alias"?

Good point, the terminology here can be easily improved. "good" -> "unconflicted".

> How would a client know what to do with this error?

Send the stack trace/error to us? If we have this specific error text, we can instantly pinpoint where the code exploded.
I guess as long as we have a way to identify the cause of the error, that's enough. Would it be better to provide an opaque error to the client? Or panic because this is clearly an unexpected scenario?

> I really don't think we should have this check. We are performing an existence check on a list->map we created by reading from the map one line before the loop. It does not make the code easier to read.

I don't like the idea of ignoring errors, even if currently **know** that these errors are impossible. What if we introduce a bug into `findConflicts`? Assuming that the lookup will be successful will cause issues.

I had to make [this change](https://github.com/dolthub/go-mysql-server/pull/1633/files#diff-d9767d452532ba5195b8da77ef173608c57601567100f4d947bf12e513a5b7beR291) because [this lookup](https://github.com/dolthub/go-mysql-server/blob/pavel/dolt-5342/sql/analyzer/hoist_select_exists.go#L308) was failing. That was happening because we weren't properly handling `Table` ambiguity, just `TableAlias` ambiguity.